vavr/src/test/java/io/vavr/MatchTest.java

```diff
@@ -318,12 +319,12 @@ public void shouldDecomposeListOfTuple3() {
     @SuppressWarnings("UnnecessaryLocalVariable")
     @Test
     public void shouldDecomposeListWithNonEmptyTail() {
-        final List<Option<Number>> intOptionList = List.of(Option.some(1), Option.some(2.0));
-        final String actual = Match(intOptionList).of(
+        final List<Option<? extends Number>> numberOptionList = List.of(Option.some(1), Option.some(2.0));
```

Comment:
I think because of the added `? extends` the build currently breaks. Does it not compile with jdk9 otherwise?
Another fix could be `List.<Option<Number>> of(Option.some(1), Option.some(2.0));`

Reply:
Dunno, man, this seems to randomly pass or fail on `JDK 9`. There's a wedding I should attend, instead I'm debugging this :p

Reply:
Go to the medding, wan - eh I mean wedding, man. We can do this later :)

Reply:
darn ... could you take a look, it's hopeless (it just passed locally ...)?

Reply:
Yes, will do that later today! Currently I'm in the middle of splitting #2093 in to smaller PRs. After that (and after going to german election)...